python/ray/serve/_private/deployment_state.py

```diff
@@ -762,16 +769,31 @@ def start(self, deployment_info: DeploymentInfo, version: DeploymentVersion):
         self._prev_slow_startup_warning_time = time.time()
         self._version = version
 
-    def update_user_config(self, user_config: Any):
+    def reconfigure(self, version: DeploymentVersion):
         """
-        Update user config of existing actor behind current
+        Update deployment config of existing actor behind current
         DeploymentReplica instance.
         """
-        self._actor.update_user_config(user_config)
-        self._version = DeploymentVersion(
-            self._version.code_version, user_config=user_config
+        # Update options from deployment config that are tracked in ActorReplicaWrapper
+        self._actor._max_concurrent_queries = (
+            version.deployment_config.max_concurrent_queries
+        )
+        self._actor._graceful_shutdown_timeout_s = (
+            version.deployment_config.graceful_shutdown_timeout_s
+        )
+        self._actor._health_check_period_s = (
+            version.deployment_config.health_check_period_s
+        )
+        self._actor._health_check_timeout_s = (
+            version.deployment_config.health_check_timeout_s
         )
```

Comment:
we do we need to copy each of these fields individually instead of just having a copy of `deployment_config`?

Reply:
we shouldn't be reaching into private fields if `self._actor` here

Reply:
Good point, I've simplified things a bit!